adr/2021_04_15_using_iri_instead_of_code_id.md

```diff
@@ -0,0 +1,46 @@
+# Using iri instead of code/id in requests
+
+* Status: proposed
+* Date: 2021-04-15
+
+## Context and Problem Statement
+
+ApiPlatform provides and recommends using IRI as identifier. That identifier is more clearly than classic id and 
+contain more information. On resources, ApiPlatform handles IRI out of the box.
+During creation our new API in many places we decided to use commands while creating our endpoints. For certain cases
+this solution is more flexible than classic approach, but it doesn't handle transformation IRI to proper `code`/`id`.
+In the past we decided to use `code`/`id` instead of IRI. Now we try to unify API and replace this approach for solution with using IRI.
+Main problem is sent request with field as IRI and creation command without IRI, but with `id`/`code`. 
+It's needed for proper handling behaviour in command handler.
```

Comment:
```suggestion
API Platform recommends using IRI as an identifier. That identifier gives more clarity than an id because it contains more information - a full endpoint path to the resource and its unique identifier. On resources, API Platform handles IRI out of the box.
While we were designing Sylius new API, we have decided to use commands in many endpoints. 
In some cases, this solution is more flexible than the default creation approach (as we have control over what we are processing), but it doesn't handle the transformation of IRI to proper `code`/`id`.

In the past, we have been using `code`/`id` instead of IRI, then we have been using both approaches. 
Now we are trying to unify the new API and replace codes and ids with IRI everywhere.

The main challenge is the usage of IRI in requests, where we want to have IRI in the request but `id`/`code`in its commands and command handlers.
```